Lib/test/test_capi/test_opt.py

```diff
@@ -793,6 +793,53 @@ def testfunc(n):
         """))
         self.assertEqual(result[0].rc, 0, result)
 
+    def test_const_eliminate_function_guards(self):
+        """Note: this function must be executed in a subprocess, because when other tests are run,
+        the globals of this module might get affected, causing globals to constant promotion
+        to fail.
+        """
```

Comment:
I betcha yo ucan do without a subprocess. Instead, just create a namespace and use that, like several other tests do. The namespace directory acts as the globals for the function defined in it. Try it!

Reply:
Sorry my Python is actually somewhat rusty -- which namespace are you referring to? I don't remember `types.SimpleNamespace` being that powerful.

Reply:
Just a dict -- look at `test_type_inconsistency()` in the same file for a complete example. Because it says `exec(src, ns, ns)` the `ns` dict is used as the globals for the function, and you can add/update its "globals" using `ns["some_name"] = some_value`.